mailpoet/lib/Config/ServicesChecker.php

```diff
@@ -161,4 +165,12 @@ public function getAnyValidKey(): ?string {
     }
     return null;
   }
+
+  public function generateHalfApiKey() {
```

Comment:
I am not sure if the name is ideal. The fact that the key is split in the middle is only an implementation detail. I think the important part here is that the key is partial, would the better name be `generatePartialApiKey`?

Also, why don't you use type hints here?

Reply:
@pavel-mailpoet Noted 👍.